src/core/datastore/sync/sync-manager.js

```diff
@@ -347,4 +352,66 @@ export class SyncManager {
         return Promise.reject(new SyncError('Invalid sync event name'));
     }
   }
+
+  _getPullLimit(pullOptions) {
+    const limitOption = pullOptions && pullOptions.pullLimit;
+    return limitOption || Infinity;
+  }
+
+  _getInternalPullQuery({ filter, sort, fields }, pullLimit) {
```

Comment:
I don't understand the significance of this method. We're going to call `_getPaginationQueries` anyway, so creating this `pullQuery` with meaningless skip and limit seems unnecessary.
Nothing wrong with the method per se, but it'll be cleaner to just create all the pagination queries in one place.

Reply:
It isn't a pagination query, they are all created in one place. There was some back and forth with building/sanitizing of queries.

Its goal was to "sanitize" the user query and make it so that all internal pagination methods can use the queries being passed to them safely. It used to add the implicit sort, but looks like we're not going that route. Now its main purpose is adding the limit, when the user supplies it in the pullOptions. This is important for the local delete(), in order to avoid deleting everything.

I may rearrange things a bit today.